x/programs/test/programs/call_program/src/lib.rs

```diff
@@ -51,12 +52,16 @@ pub fn call_with_two_params_external(
     value1: i64,
     value2: i64,
 ) -> i64 {
-    let args: Vec<_> = value1
-        .to_le_bytes()
-        .into_iter()
-        .chain(value2.to_le_bytes())
-        .collect();
+    let args: Vec<_> = borsh::to_vec(&(value1, value2)).expect("serialization failed");
     target
         .call_function("call_with_two_params", &args, max_units)
         .unwrap()
 }
+
+#[public]
+pub fn call_deferred(_: Context, target: Program, max_units: Gas) -> i64 {
+    let bytes = target
+        .call_function::<DeferDeserialize>("simple_call", &[], max_units)
+        .unwrap();
+    bytes.deserialize()
+}
```

Comment:
You don't need an integration test to test this feature, it can be done with a unit test.